homeassistant/components/sensor/ebusd.py

```diff
@@ -0,0 +1,185 @@
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+SCAN_INTERVAL = timedelta(seconds=30)
+
+SENSOR_TYPES = {
+    'hc1ActualFlowTempDesired': ['Hc1ActualFlowTempDesired','°C', 'mdi:thermometer', 'decimal'],
+    'hc1MaxFlowTempDesired': ['Hc1MaxFlowTempDesired', '°C', 'mdi:thermometer', 'decimal'],
+    'hc1MinFlowTempDesired': ['Hc1MinFlowTempDesired', '°C', 'mdi:thermometer', 'decimal'],
+    'hc1PumpStatus': ['Hc1PumpStatus', None, 'mdi:toggle-switch', 'switch'],
+    'hc1SummerTempLimit': ['Hc1SummerTempLimit', '°C', 'mdi:weather-sunny', 'decimal'],
+    'holidayTemp': ['HolidayTemp', '°C', 'mdi:thermometer', 'decimal'],
+    'hwcTempDesired': ['HwcTempDesired', '°C', 'mdi:thermometer', 'decimal'],
+    'hwcTimerMonday': ['hwcTimer.Monday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerTuesday': ['hwcTimer.Tuesday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerWednesday': ['hwcTimer.Wednesday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerThursday': ['hwcTimer.Thursday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerFriday': ['hwcTimer.Friday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerSaturday': ['hwcTimer.Saturday', None, 'mdi:timer', 'time-schedule'],
+    'hwcTimerSunday': ['hwcTimer.Sunday', None, 'mdi:timer', 'time-schedule'],
+    'waterPressure': ['WaterPressure', 'bar', 'mdi:water-pump', 'decimal'],
+    'z1RoomZoneMapping': ['z1RoomZoneMapping', None, 'mdi:label', 'decimal'],
+    'z1NightTemp': ['z1NightTemp', '°C', 'mdi:weather-night', 'decimal'],
+    'z1DayTemp': ['z1DayTemp', '°C', 'mdi:weather-sunny', 'decimal'],
+    'z1HolidayTemp': ['z1HolidayTemp', '°C', 'mdi:thermometer', 'decimal'],
+    'z1RoomTemp': ['z1RoomTemp', '°C', 'mdi:thermometer', 'decimal'],
+    'z1ActualRoomTempDesired': ['z1ActualRoomTempDesired', '°C', 'mdi:thermometer', 'decimal'],
+    'z1TimerMonday': ['z1Timer.Monday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerTuesday': ['z1Timer.Tuesday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerWednesday': ['z1Timer.Wednesday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerThursday': ['z1Timer.Thursday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerFriday': ['z1Timer.Friday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerSaturday': ['z1Timer.Saturday', None, 'mdi:timer', 'time-schedule'],
+    'z1TimerSunday': ['z1Timer.Sunday', None, 'mdi:timer', 'time-schedule'],
+    'continuosHeating': ['ContinuosHeating', '°C', 'mdi:weather-snowy', 'decimal'],
+    'prEnergySumHcLastMonth': ['PrEnergySumHcLastMonth', 'kWh', 'mdi:flash', 'decimal'],
+    'prEnergySumHcThisMonth': ['PrEnergySumHcThisMonth', 'kWh', 'mdi:flash', 'decimal']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_CIRCUIT): cv.string,
+    vol.Optional(CONF_TIME, default=0): cv.positive_int,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[]): vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
```

Comment:
Please keep this ordered. `Required` first then `Optional`...both alphabetically. Much faster to work on it in the future.